### PR TITLE
Hygiene changes in MQTT integration tests

### PR DESCRIPTION
**Pull request by aggarw13**
Hygiene changes in MQTT integration tests

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: integration-test/mqtt/mqtt_system_test.c [hunk 1]
```diff
@@ -1123,9 +1120,6 @@ void test_MQTT_Restore_Session_Resend_PubRel( void )
                        MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
     TEST_ASSERT_FALSE( receivedPubComp );
 
-    /* Verify that the connection with the broker has been disconnected. */
-    TEST_ASSERT_EQUAL( MQTTNotConnected, context.connectStatus );
-
```
Comment (muneebahmed10) [inline integration-test/mqtt/mqtt_system_test.c:1128]: In the other tests, we reset `disconnectOnPacketType` back to `MQTT_PACKET_ID_INVALID` after resuming the session. I think we should do the same here, even if it doesn't affect the test.

**Review by muneebahmed10 — COMMENTED** [review]
Seems kind of late to be making a change to integration tests, especially since it changes the code. Will this change also be going into release candidate?

File: integration-test/mqtt/mqtt_system_test.c (hunk 1 above)
Comment (aggarw13) [reply integration-test/mqtt/mqtt_system_test.c:1129]: Good suggestion. Done.

File: integration-test/mqtt/mqtt_system_test.c [hunk 2]
```diff
@@ -1145,7 +1143,7 @@ void test_MQTT_Restore_Session_Resend_PubRel( void )
  * incoming QoS 2 PUBLISH operation that was incomplete in a previous connection
  * of the same session.
  */
-void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
+void test_MQTT_Restore_Session_Complete_Duplicate_Incoming_PubRel( void )
```
Comment (muneebahmed10) [inline integration-test/mqtt/mqtt_system_test.c:1146]: "Complete duplicate incoming pubrel"?

Comment (aggarw13) [reply integration-test/mqtt/mqtt_system_test.c]: Removed "Complete" from the name

**Review by muneebahmed10 — APPROVED** [review]

File: integration-test/mqtt/mqtt_system_test.c [hunk 3]
```diff
@@ -574,12 +574,9 @@ static void eventCallback( MQTTContext_t * pContext,
     TEST_ASSERT_EQUAL( MQTTSuccess, pDeserializedInfo->deserializationResult );
     pPublishInfo = pDeserializedInfo->pPublishInfo;
 
-    if( ( pPacketInfo->type == disconnectOnPacketType ) ||
-        ( ( pPacketInfo->type & 0xF0U ) == disconnectOnPacketType ) )
+    if( ( pPacketInfo->type == packetTypeForDisconnection ) ||
+        ( ( pPacketInfo->type & 0xF0U ) == packetTypeForDisconnection ) )
     {
-        /* Terminate MQTT connection with server for session restoration test. */
-        TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
```
Comment (dan4thewin) [inline integration-test/mqtt/mqtt_system_test.c:581]: Was this moved?

File: integration-test/mqtt/mqtt_system_test.c [hunk 4]
```diff
@@ -1119,17 +1116,18 @@ void test_MQTT_Restore_Session_Resend_PubRel( void )
...
     /* Disconnect on receiving PUBREC so that we are not able to complete the QoS 2 PUBLISH in the current connection. */
     TEST_ASSERT_FALSE( receivedPubComp );
-    disconnectOnPacketType = MQTT_PACKET_TYPE_PUBREC;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_PUBREC;
     TEST_ASSERT_EQUAL( MQTTSendFailed,
                        MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
     TEST_ASSERT_FALSE( receivedPubComp );
 
-    /* Verify that the connection with the broker has been disconnected. */
-    TEST_ASSERT_EQUAL( MQTTNotConnected, context.connectStatus );
```
Comment (dan4thewin) [inline integration-test/mqtt/mqtt_system_test.c:1128]: Why was this removed?

File: integration-test/mqtt/mqtt_system_test.c [hunk 5]
```diff
@@ -1170,20 +1168,17 @@ void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
 
     /* Disconnect on receiving PUBREL so that we are not able to complete in the incoming QoS2
      * PUBLISH in the current connection. */
-    disconnectOnPacketType = MQTT_PACKET_TYPE_PUBREL;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_PUBREL;
     TEST_ASSERT_EQUAL( MQTTSendFailed,
                        MQTT_ProcessLoop( &context, 3 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
 
-    /* Verify that the connection with the broker has been disconnected. */
-    TEST_ASSERT_EQUAL( MQTTNotConnected, context.connectStatus );
```
Comment (dan4thewin) [inline integration-test/mqtt/mqtt_system_test.c:1178]: ?

File: integration-test/mqtt/mqtt_system_test.c (hunk 3 above)
Comment (aggarw13) [reply integration-test/mqtt/mqtt_system_test.c:581]: This was removed as the `MQTT_Disconnect` call is not explicitly made to terminate the MQTT connection, but the network connection is severed directly by calling `OpenSsl_Disconnect`.

Only terminating the network connection is required for terminating an MQTT session

File: integration-test/mqtt/mqtt_system_test.c (hunk 4 above)
Comment (aggarw13) [reply integration-test/mqtt/mqtt_system_test.c:1128]: As `MQTT_Disconnect` call has been removed, the `MQTTContext.connectedStatus` would not be updated

File: integration-test/mqtt/mqtt_system_test.c (hunk 5 above)
Comment (aggarw13) [reply integration-test/mqtt/mqtt_system_test.c:1178]: Same as [this](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1252#discussion_r489106292)

**Review by dan4thewin — APPROVED** [review]
